### DISCUSSION TITLE
Before you graduate!

### DISCUSSION THREAD

**Discussion by onlyphantom** [teamalgoritma/community#121] [Tips]
### Who am I writing this for

This guide is especially written for anyone taking their first step into becoming a **data science professional**. In the [Data Science Fun to Profit workshop](https://www.facebook.com/groups/datasciencetutorials/permalink/5523157821088896/), I set out some definitions on what qualifies as professional. 

> Professional gets paid for delivering a service, or a product

So as you graduate from the bootcamp, you're entering a new chapter where you no longer write code to complete assignments or learn-by-building modules. You'll soon be writing code to solve problems, whether in the form of a service or product, and more excitingly, **get paid for doing that**.

### Your first steps

1. Algoritma Alumni Slack channel: Join the channel and participate in questions / answers. Network with others. 
1. Your GitHub profile: Start your portfolio early! Pin the work you're most proud of so they are featured on your profile page, and optionally create a [special GitHub Profile README](https://youtu.be/JIxRaOQfxAo) to show others where they can connect with you
1. Equally important is a LinkedIn profile. If you're new to LinkedIn, you can start by connecting with your course mates, your instructors and teaching assistants. Having a LinkedIn presence increases your chance of getting job offers and interviews.   
1. Write a [good, parseable, semantic CV and keep visual elements like donut chart, bar charts etc down to a minimum](https://youtu.be/gpr8uZHSkcA). If you need help with this, or would like feedback on this, reach out to me (details at the bottom)!
1. Put your work in the spotlight. It makes no sense to put in tens of hours of sweat and tears making a really cool Shiny app, or Flask app, only to have them watched by 10 people in total. That's not doing your talent and effort any justice. 
    - Share it with your connections on LinkedIn, pin it to your GitHub profile, write a Medium article about it, talk to anyone who asked about it, create a 30-second clip to introduce your project, get on a podcast to talk about it, ask for my help to share it
1. Join [data science communities](https://www.facebook.com/groups/datasciencetutorials) and participate in discussions
1. If you're an alumni of Algoritma's Lifetime Academy, use your [lifetime workshop privileges](https://algorit.ma/explore-academy/) to sign up for all our monthly workshops for free, for a lifetime. Applies to all Data Science Series workshops (regular price of IDR2-3m), Kickstart workshops (regular price of IDR100k), and all other short-form courses we'll ever launch.
1. StackOverflow may look like a hostile, intimidating environment for new beginners but learning how to ask questions and get feedback could be a very valuable experience. Sign up for a StackOverflow account and learn the etiquette to make the most out of the site.

### General Advice
> [Feeling stupid](https://blogs.oregonstate.edu/gemmlab/2018/11/05/why-feeling-stupid-is-great-how-stupidity-fuels-scientific-progress-and-discovery/) is a leading indicator of knowledge discovery

Do not feel shy about asking question or putting your work out there for fear of looking incompetent. 

Also, there's a beauty about quantity when you're starting out. Beginners tend to over-emphasize on quality on work and under-estimate the importance of quantity. When you're only starting to write code, you have no established mental models that reinforce good habits, or proven mental models that you know are great fit to any given problem. Putting too much focus on quality is only going to frustrate you as the perfectionism may serve to hinder your progress instead of propel it. 

My suggestion is to produce work and let the sheer quantity work in your favor. Accept that your app is never going to be _flawless_, perfect, due to your inexperience. Honestly, I look back at code I've written just a year ago and I sometimes cringe. As a beginner, our first few apps are going to be _bad_, but that's OK because it's part of the process. 

Finally, learn how to learn faster. Spend one day every week to learn the shortcuts of the top 2-3 software you use most. Optimize your workflow. Automate common tasks. 

### Stay in touch
This guide is probably never going to be "done". We'll treat it as a living document and anyone is welcome to edit and add commentary to it. I'll come back to it when I have new ideas to add.

If you need help but would like to message me privately, you can do so on:
- [LinkedIn](https://id.linkedin.com/in/chansamuel)
- [Twitter](https://twitter.com/_onlyphantom)
- Email: `paste("samuel", "@", "algorit.ma", sep="")` 

I'm happy to take a call with you to answer any questions you have (I speak very limited Bahasa Indonesia) or to help you in any way as you prepare to take your first step into the real world. 
Good luck!

**Comment by samuelmulia** [comment]
Hi Guys,

I put together some tips to help you in your career search. You can download the deck from the following link: [Google Drive](https://drive.google.com/file/d/1tiqUuLH8ESnv1KAJyw7LMcfiUwVin_br/view?usp=sharing). This deck discusses how you can improve personal branding on Linkedin, the activities you can do, and how to get more interview opportunities.

I hope this information is helpful for you, and feel free to reach me at career.center@algorit.ma if you need any help in your career search.

Best of luck,
Mulia
